### ISSUE TITLE
[Feature] Allow to set test.use params per single test

### ISSUE THREAD

**Issue by AdamOakman** [microsoft/playwright#27138]
This issue was mentioned in a couple of places and seems like it is missing a good solution: https://github.com/microsoft/playwright/discussions/19893
https://github.com/microsoft/playwright/discussions/21341

Currently we have 2 ways of using fixture options - either by setting it in the config file per project inside `use` block or in `test.use` statement in a spec file. `test.use` statement seems to be not flexible enough because it allows to set whatever we want only for the whole file or per whole `test.describe` statement and not for a single test.

Taking simple example from the official documentation https://playwright.dev/docs/test-fixtures#fixtures-options - having fixture option defined like below:

```ts
import { test as base } from '@playwright/test';
import { TodoPage } from './todo-page';

// Declare your options to type-check your configuration.
export type MyOptions = {
  defaultItem: string;
};
type MyFixtures = {
  todoPage: TodoPage;
};

// Specify both option and fixture types.
export const test = base.extend<MyOptions & MyFixtures>({
  // Define an option and provide a default value.
  // We can later override it in the config.
  defaultItem: ['Something nice', { option: true }],

  // Our "todoPage" fixture depends on the option.
  todoPage: async ({ page, defaultItem }, use) => {
    // ...
  },
});
export { expect } from '@playwright/test';
```

It would be convenient to be able to set use params per single test like that (or any other neat way that would do the trick):

```ts
test('example test #1', async ({ todoPage, page }) => {
  // ...
}).use({ defaultItem: 'Something cool' });

test('example test #2', async ({ todoPage, page }) => {
  // ...
}).use({ defaultItem: 'Something else' });
```

**Comment by Terence625** [comment]
yes, I was trying to use multiple `test.use()` to switch authentications between tests in one `test.describe` block, but the last `test.use()` will just override the previous ones. And those tests are in one `test.describe` block for a reason as they are same tests with different authentications, I don't want to separate them into different files or different projects. It'll be nice to have `test.use()` more flexible

**Comment by phil294** [comment]
>  but the last test.use() will just override the previous ones.

This is also my observation. I wonder what is the reasoning for the current behavior? We'd like to run a few small tests across different locales, nested under other conditionals. Apparently this is currently impossible? I tried

```js
myLocales.forEach(locale => {
    const test = base.extend({})
    test.use({locale})
    test('...')
})
```
but apparently even like that, every test uses the same locale, namely the *last* one specified in `myLocales`. Don't really understand why though

**Comment by kenanAlhindi97** [comment]
I don't know if I should make a separate issue for this but one of the behaviors that prevents from using fixtures is the fact that :
`test.use({parameter:parameterValue});` 
is not editable/update-able 

say you have a this code : 

`
let parameterValue=foo;
test.beforeEach(async () => {
    paremeterValue=bar;
});
test.use({parameter:parameterValue});

test('should log bar', async ({parameter}) => {
   console.log(parameter) // 'foo'
});
`
I just don't see any other way to implement parameterized fixtures with mutable parameters, and it would be a nice feature.

**Comment by NagRock** [comment]
It would be nice if test could override configured options with the one provided as last param:

```test('should log bar', async ({parameter}) => {
   console.log(parameter) // 'foo'
}, () => ({parameter: parameterValue}));```

And the function with overrides will be called just before test function to override parent configuration

**Comment by NoamGaash** [comment]
Just asking - what's bad with empty anonymous describe blocks?
For example:
```js
test.describe(() => {
  test.use({locale: 'he-IL'})
  test('validate the hebrew version of the website', async ({page, eyes}) => {
    await page.goto('https://example.com')
    await eyes.check()
  })
})
```

**Comment by vitalets** [comment]
> Just asking - what's bad with empty anonymous describe blocks?

I see two points:
1. syntax is more cumbersome
2. VSCode does not show it well https://github.com/microsoft/playwright/issues/30476

**Comment by Zirnael** [comment]
I ended up writing a helper for myself to avoid the cumbersome syntax.

```
export function testWith(
  title: string,
  options: Partial<FixtureOptions>,
  testFn: (args: FixtureOptions & SetupMethod) => Promise<void>,
) {
  test.describe(() => {
    test.use(options);
    test(title, testFn);
  });
}
```

Example fixture:
```
export type TestUsers =
  | 'Almighty'
  | 'Operator'

export type FixtureOptions = {
  user: TestUsers;
};

export const test = baseTest.extend<FixtureOptions>({
  user: ['Almighty', { option: true }],
});
```

Usage:
```
testWith('Test with operator user', { user: 'Operator' }, async ({ page }) => {
  await expect(page).toHaveScreenshot("example.png");
});

```

**Comment by vitalets** [comment]
> I ended up writing a helper for myself to avoid the cumbersome syntax.

That's nice. I'm curious how does it work with vscode extension - can I run a particular test?

**Comment by Mischkedb** [comment]
This would be a fantastic feature. WE want to be able to run more small tests in parallel where for example we have several tests that update a user's name in a form. One saves, one undos. etc etc. It would be good if we could use different users for each of these small tests that fall under a single describe. We use storage states and I would love to soecidy test.use inside test.

**Comment by Gabriela-Rogowska** [comment]
This feature would be extremely helpful in my current workflow. Right now the lack of functionality forces us to nest describes, which causes duplications in my code and more complex structure of reports. 
I’d really appreciate seeing this added to Playwright — thanks for all your great work!

**Comment by dbarrientostk** [comment]
any updates?
